core/src/main/scala/io/udash/properties/seq/ForwarderReadableSeqProperty.scala

```diff
@@ -9,34 +9,50 @@ private[properties] trait ForwarderReadableSeqProperty[A, B, ElemType <: Readabl
 
   protected def origin: ReadableSeqProperty[A, OrigType]
 
-  private var originListenerRegistration: Registration = _
+  protected final var transformedElements: MBuffer[ElemType] = CrossCollections.createArray[ElemType]
   private var originStructureListenerRegistration: Registration = _
 
-  protected final def initialized: Boolean = originListenerRegistration != null && originListenerRegistration.isActive
+  protected final def initialized: Boolean = originStructureListenerRegistration != null && originStructureListenerRegistration.isActive
+
+  protected def loadFromOrigin(): BSeq[B]
+  protected def elementsFromOrigin(elemProperties: BSeq[OrigType]): BSeq[ElemType]
+  protected def transformPatchAndUpdateElements(patch: Patch[OrigType]): Opt[Patch[ElemType]]
+
+  protected def originStructureListener(patch: Patch[OrigType]): Unit =
+    transformPatchAndUpdateElements(patch).foreach { transformed =>
+      structureListeners.foreach(_.apply(transformed))
+      valueChanged()
+    }
+
+  protected def onListenerInit(originElems: BSeq[OrigType]): Unit = {
```

Comment:
This is intentional. A lot of the bugs in property implementations came from the fact, that `origin.elemProperties` is not idempotent in case of more complex transformations in some situations and we forgot to cache the first call.